lib/content-services/src/lib/content-node-share/content-node-share.dialog.ts

```diff
@@ -266,25 +274,46 @@ export class ShareDialogComponent implements OnInit, OnDestroy {
         this.form.setValue({
             sharedUrl: `${this.baseShareUrl}${this.sharedId}`,
             time: expiryDate ? moment(expiryDate).local() : null
-        });
+        }, { emitEvent: false });
 
-        if (expiryDate) {
-            this.time.enable();
-        } else {
-            this.time.disable();
-        }
+        return expiryDate;
     }
 
-    private updateNode(date: moment.Moment): Observable<Node> {
+    private updateNode(date: moment.Moment) {
+
         const expiryDate = date
-            ? (this.type === 'date' ? date.endOf('day').utc().format() : date.utc().format())
+            ? (this.type === 'date' ? date.endOf('day').utc().format('YYYY-MM-DDTHH:mm:ss.SSSZ')
+            : date.utc().format('YYYY-MM-DDTHH:mm:ss.SSSZ'))
```

Comment:
Can we extract this nested ternary operation into an independent statement as sonarcloud suggests?

Reply:
yess, i resolved this nested ternary operation into an independent statement.